tools/src/launcher/Main.hs

```diff
@@ -559,14 +629,18 @@ system'
     -- ^ Command
     -> [Text]
     -- ^ Lines of standard input
+    -> NodeType
+    -- ^ node/wallet log output
     -> io ExitCode
     -- ^ Exit code
-system' phvar p sl = liftIO (do
+system' phvar p sl nt = liftIO (do
     let open = do
-            (m, _, _, ph) <- Process.createProcess p
+            (m, stdO, stdE, ph) <- Process.createProcess p
             putMVar phvar ph
             case m of
-                Just hIn -> IO.hSetBuffering hIn IO.LineBuffering
+                Just hIn -> do
+                    _ <- asyncLog (fromJust stdO) (fromJust stdE) ph nt
```

Comment:
Wait, AFAIU, it means that now `system'` will apply `asyncLog` logic to the spawned process every time, unconditionally. Are we sure this is what we want? What if we spawn process with `std_out = Inherit`? We will read out own output and print it our output, then read again, etc. Won't it lead to infinite loop?
Anyway, even if it won't, I don't see any reason to do it.
There is only one case when we want to apply `asyncLog` logic: when want to print output of both node and wallet.

Reply:
Also are you sure that `std0` and `stdE` will always be `Just`? Can you write a comment explaining why it's true. It's not obvious from this code, so it's better to have a comment explaining that.

Reply:
AFAIU, it only makes sense to use `asyncLog` when the process is spawned with `std_out = CreatePipe` and `std_err = CreatePipe` and maybe `UseHandle`.
If it's `Inherit`, there will be infinite loop probably. If it's `NoStream`, `std0` and `stdE` will be `Nothing` most likely.

Reply:
Right, I am checking now what constructor is used to create `StdStream`